### PR TITLE
refactor!: removal of deprecated code from V24

### PR DESCRIPTION
**Pull request by MarcinVaadin**
Fixes: #15665

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/AppShellRegistry.java [hunk 1]
```diff
@@ -57,12 +57,6 @@ public class AppShellRegistry implements Serializable {
     public static final String ERROR_HEADER_OFFENDING_PWA = "%n%nWe changed the way you configure PWAs, please see this link for more info:%n"
             + "https://vaadin.com/docs/latest/flow/advanced/modifying-the-bootstrap-page%n";
 
-    public static final String ERROR_HEADER_NO_APP_CONFIGURATOR = "%n%nThe `PageConfigurator` interface is deprecated since Vaadin 15 and has no effect."
-            + "%nPlease, create a class implementing `AppShellConfigurator`, and remove `PageConfigurator` from: %n  - %s%n";
-
-    public static final String ERROR_HEADER_OFFENDING_CONFIGURATOR = "%n%nThe `PageConfigurator` interface is deprecated since Vaadin 15 and has no effect."
-            + "%nPlease, configure the page in %s, and remove the `PageConfigurator` from: %n - %s%n";
-
     private static final String ERROR_LINE = "  - %s from %s";
```
[the remark this answers is not in the document]
Comment (MarcinVaadin) [reply flow-server/src/main/java/com/vaadin/flow/server/AppShellRegistry.java:60]: @sonatype-lift ignore

**Review by tepi — DISMISSED** [review]
Looks good to me, although I'm not surely familiar with all the details.

File: vaadin-spring/src/test/java/com/vaadin/flow/spring/MockVaadinContext.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    public MockVaadinContext(ServletContext context,
+            RoutePathProvider provider) {
+        super(context);
+
+        Mockito.when(lookup.lookup(RoutePathProvider.class)).thenReturn(null);
+
+        Mockito.when(lookup.lookup(RoutePathProvider.class))
+                .thenReturn(provider);
```
Comment (mcollovati) [inline vaadin-spring/src/test/java/com/vaadin/flow/spring/MockVaadinContext.java:60]: Does this mean that `lookup.lookup(RoutePathProvider.class)` should always return null at first call and then the given provider?
If so we can compact it in a single instruction
```java
Mockito.when(lookup.lookup(RoutePathProvider.class)).thenReturn(null, provider)
```

File: flow-server/src/main/java/com/vaadin/flow/di/Instantiator.java [hunk 3]
```diff
@@ -37,37 +37,13 @@
  * that manages instances according to the conventions of that framework.
  * <p>
  * {@link VaadinService} will by default use {@link ServiceLoader} for finding
- * an instantiator implementation. Deployment will fail if multiple candidates
- * are returning <code>true</code> from {@link #init(VaadinService)}. If no
- * candidate is found, {@link DefaultInstantiator} will be used. It is possible
- * to override this mechanism by overriding
```
[the remark this answers is not in the document]
Comment (MarcinVaadin) [reply flow-server/src/main/java/com/vaadin/flow/di/Instantiator.java:43]: Yes, I've brought back DefaultInstantiator part leaving removed only part about multiple instantiators present at once - not it is not allowed.

File: vaadin-spring/src/test/java/com/vaadin/flow/spring/MockVaadinContext.java (hunk 2 above)
Comment (MarcinVaadin) [reply vaadin-spring/src/test/java/com/vaadin/flow/spring/MockVaadinContext.java:60]: TBH I've just copied that class from flow-server not getting into details.

Comment (MarcinVaadin) [reply vaadin-spring/src/test/java/com/vaadin/flow/spring/MockVaadinContext.java:60]: I've checked and tests are failing after your proposal so I'm leaving it as is for the moment.

**Review by mcollovati — APPROVED** [review]

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/security/VaadinWebSecurity.java [hunk 4]
```diff
@@ -415,7 +425,16 @@ protected void setLoginView(HttpSecurity http,
                             + flowLoginView.getName());
         }
 
-        String loginPath = RouteUtil.getRoutePath(flowLoginView, route.get());
+        if (!(applicationContext instanceof WebApplicationContext)) {
+            throw new RuntimeException(
+                    "VaadinWebSecurity cannot be used without WebApplicationContext.");
+        }
+
+        VaadinServletContext vaadinServletContext = new VaadinServletContext(
```
[the remark this answers is not in the document]
Comment (MarcinVaadin) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/security/VaadinWebSecurity.java:433]: @sonatype-lift ignoreall

**Pull request merged**